src/controllers/user.controller.js

```diff
@@ -1,22 +1,47 @@
 import { UserService } from '../services/user.service.js';
 import { BcryptUtility } from '../utils/bcrypt.util.js';
 import { JwtUtility } from '../utils/jwt.util.js';
+import { User } from '../database/models/index';
 
-export class RegisterController {
-  static async registerUser(req, res) {
-    try {
-      const user = { ...req.body };
-      user.password = await BcryptUtility.hashPassword(req.body.password);
-      const { id, email } = await UserService.register(user);
-      const userData = { id, email };
-      const userToken = JwtUtility.generateToken(userData);
-      return res.status(201).json({ user: userData, token: userToken });
-    } catch (err) {
-      return res.status(500).json({
-        error: err.message,
-        message: 'Failed to register a new user',
-      });
-    }
-  }
+export class UserController {
+	static async registerUser(req, res) {
+		try {
+			const user = { ...req.body };
+			user.password = await BcryptUtility.hashPassword(req.body.password);
+			const { id, email } = await UserService.register(user);
+			const userData = { id, email };
+			const userToken = JwtUtility.generateToken(userData);
+			return res.status(201).json({ user: userData, token: userToken });
+		} catch (err) {
+			return res.status(500).json({
+				error: err.message,
+				message: 'Failed to register a new user',
+			});
+		}
+	}
+
+	static async loginUser(req, res) {
+		try {
+			const user = await User.findOne({
+				where: { email: req.body.email },
+			});
+			const isValidPassword = await BcryptUtility.verifyPassword(
+				req.body.password,
+				user.password
+			);
+			if (!isValidPassword)
+				return res.status(400).json({ error: 'Wrong credentials' });
+
+			const userData = { id: user.id, email: user.email, role: user.role };
+			const token = JwtUtility.generateToken(userData);
+			return res.status(200).json({
+				token: token,
+				message: 'Login successful',
+				email: user.email,
+			});
+		} catch (err) {
+			return res.status(400).json({ error: err });
+		}
+	}
```

Comment:
Follow the same convention implement when registering the user. Check in the controller or ask @Abdden